app/src/features/apiClient/commands/collectionRunner/runCollection.command.ts

```diff
@@ -181,8 +186,36 @@ class Runner {
     });
   }
 
+  private populateAutogenerateStore(recordId: string) {
+    console.log("called");
+    const { getData, getParentChain } = this.ctx.stores.records.getState();
+    const apiRecord = getData(recordId);
+
+    if (!apiRecord || !apiRecord.data) {
+      return;
+    }
+
+    const childDetails = {
+      id: apiRecord.id,
+      parentId: apiRecord.collectionId,
+    };
+
+    const resolvedEntry = renderVariables(apiRecord.data, apiRecord.id, this.ctx);
```

Comment:
Do we really need to resolve the entire entry object here? Cant we only resolve the stuff needed to populate the autogenerate store?
This is because, preparation service would again resolve the variables as in preparation service some other keys would be added/modified which makes it necessary to prepare over there. So, redundant variable's resolution may be remvoed from here